apps/segment/src/modules/db/segment-main-table.ts

```diff
@@ -0,0 +1,90 @@
+import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
+import { Entity, schema, string, Table } from "dynamodb-toolbox";
+
+import { createDynamoDBClient, createDynamoDBDocumentClient } from "@/lib/dynamodb-client";
+
+export class SegmentMainTable extends Table<
+  { name: "PK"; type: "string" },
+  {
+    name: "SK";
+    type: "string";
+  }
+> {
+  private constructor(
+    args: ConstructorParameters<
+      typeof Table<
+        { name: "PK"; type: "string" },
+        {
+          name: "SK";
+          type: "string";
+        }
+      >
+    >[number],
+  ) {
+    super(args);
+  }
+
+  static create({
+    documentClient,
+    tableName,
+  }: {
+    documentClient: DynamoDBDocumentClient;
+    tableName: string;
+  }): SegmentMainTable {
+    return new SegmentMainTable({
+      documentClient,
+      name: tableName,
+      partitionKey: { name: "PK", type: "string" },
+      sortKey: {
+        name: "SK",
+        type: "string",
+      },
+    });
+  }
+
+  static getAPLPrimaryKey({ saleorApiUrl }: { saleorApiUrl: string }) {
+    return `${saleorApiUrl}` as const;
+  }
+
+  static getAPLSortKey() {
+    return `APL` as const;
+  }
```

Comment:
Is that correct? It's "main table" but it refers to apl. I guess the class name and file name are wrong